poh-bundle-linux-x64/oh.sh

```diff
@@ -1,68 +1,197 @@
 #!/bin/bash
+#
+# Open Hospital (www.open-hospital.org)
+# Copyright © 2006-2020 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+#
+# Open Hospital is a free and open source software for healthcare data management.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# https://www.gnu.org/licenses/gpl-3.0-standalone.html
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+echo ""
+echo "***************************************************"
+echo "* Warning - Do not run this script as root user ! *"
+echo "***************************************************"
+echo ""
+
+######## Environment check:
+
+######## OPENHOSPITAL Configuration
+# POH_PATH is the directory where Portable OpenHospital files are located
+# POH_PATH=/usr/local/OpenHospital
+
+# name of this shell script
+SCRIPT_NAME="oh.sh"
+
+# set current dir
+CURRENT_DIR=$PWD
+
+if [ -z $POH_PATH ]; then
+	echo "Warning - POH_PATH not found. Please set it up properly."
+		if [ ! -f ./$SCRIPT_NAME ]; then
+		echo "Error - oh.sh not found in the current PATH. Please cd the directory where POH was unzipped or set up POH_PATH properly."
+		exit 0
+		fi
+	POH_PATH=$PWD
+fi
+
+######## Software configuration - change at your own risk :-)
 
 MYSQL_DIR="mysql-5.7.30-linux-glibc2.12-x86_64"
-JAVA_DIR="jdk8u252-b09-jre"
+MYSQL_URL="https://downloads.mysql.com/archives/get/p/23/file"
+
+JAVA_DISTRO="OpenJDK11U-jdk_x64_linux_openj9_11.0.8_10_openj9-0.21.0"
+JAVA_URL="https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.8%2B10_openj9-0.21.0"
+JAVA_DIR="jdk-11.0.8+10"
+
+
 OH_DIR="oh"
+DATABASE_NAME="oh"
+DATABASE_USER="isf"
+DATABASE_PASSWORD="isf123"
 DICOM_DEFAULT_SIZE="4M"
 
-where_i_am=$(pwd)
-cd $(dirname $0)
-POH_PATH=$(pwd)
-cd $where_i_am
+cd $POH_PATH
 
-# Find a free port to run MySQL starting from the default port.
-mysql_port=3306
-while [ $(ss -tna | awk '{ print $4 }' | grep ":$mysql_port") ]; do
-	mysql_port=$(expr $mysql_port + 1)
-done
+######################## DO NOT EDIT BELOW THIS LINE ########################
+
+echo "Checking for software...."
+
+if [ ! -d "$POH_PATH/$MYSQL_DIR" ]; then
```

Comment:
maybe split the software checks (mysql and java) into functions?
they also look very similar, might be worth to see if they can be compressed in just one function with variables, but it can be done in another change

Reply:
Yes, it might be possible, but I would rather keep it this way (at least for now) since it's more readable and I don't think we will need that function for other then mysql and java. I'll think about it anyway.